[Doc] Add developer notes

riedgar-ms opened the pull request:
Expand the CONTRIBUTING file to describe the basic developer setup. Since our workflows access some quite expensive resources, PRs have to be approved to run. This will let contributors get feedback faster.

Harsha-Nori commented:
Thoughts on extending `CONTRIBUTING.MD` with this info instead of having a new top level file here?

riedgar-ms commented:
> Thoughts on extending `CONTRIBUTING.MD` with this info instead of having a new top level file here?

That's an option too. I thought about it, but the existing file was more 'legalese' than information on how to contribute. Happy to switch if you prefer.

riedgar-ms commented on DevelopmentNotes.md:
````diff
@@ -0,0 +1,31 @@
+```
+where `<MODELNAME>` is taken from the `AVAILABLE_MODELS` dictionary defined in `conftest.py`.
+
+## Linting
````
Need to add a section on adding models to the test setup

riedgar-ms commented:
@Harsha-Nori , I've moved everything into CONTRIBUTING. I've also added some notes about how to hook new models into the testing setup on the off chance that random people start pushing to `main`.
